Added a get request to monitoring/jobs in order to enable monitor que…

amirwollman opened the pull request:
…ries.

Under this branch added two modules:
List and Retrieve.

amirwollman committed:
Added a get request to monitoring/jobs in order to enable monitor queries.
Under this branch added two modules:
List and Retrieve.

pashap commented on nsone/rest/data.py:
```diff
@@ -5,6 +5,20 @@
 #
 from . import resource
 
+class Job(resource.BaseResource):
```
Would you mind moving this class to the [rest monitoring module](https://github.com/ns1/nsone-python/blob/master/nsone/rest/monitoring.py)?

And what if we called this class `Monitors` as opposed to `Job` to keep with the convention of [Records](https://github.com/ns1/nsone-python/blob/master/nsone/rest/records.py#L9) and [Zones](https://github.com/ns1/nsone-python/blob/master/nsone/rest/zones.py#L10)?

amirwollman replied:
Yeah, sounds good to me, I'll get it done when I'm back home, probably in a
couple of hours.
On Jun 13, 2016 4:36 PM, "Pasha Palangpour" notifications@github.com
wrote:

> In nsone/rest/data.py
> https://github.com/ns1/nsone-python/pull/6#discussion_r66791928:
> 
> > @@ -5,6 +5,20 @@
> >  #
> >  from . import resource
> > 
> > +class Job(resource.BaseResource):
> 
> Would you mind moving this class to the rest monitoring module
> https://github.com/ns1/nsone-python/blob/master/nsone/rest/monitoring.py
> ?
> 
> And what if we called this class Monitors as opposed to Job to keep with
> the convention of Records
> https://github.com/ns1/nsone-python/blob/master/nsone/rest/records.py#L9
> and Zones
> https://github.com/ns1/nsone-python/blob/master/nsone/rest/zones.py#L10?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/ns1/nsone-python/pull/6/files/e9267fab90850c84ad6a25e9c3b027ce710a9c2f#r66791928,
> or mute the thread
> https://github.com/notifications/unsubscribe/AL5VfbiZH-shN4gQ5X3qnhSsry7xxPxQks5qLVzLgaJpZM4I0IES
> .

pashap commented on nsone/__init__.py:
```diff
@@ -85,6 +86,15 @@ def datafeed(self):
         import nsone.rest.data
         return nsone.rest.data.Feed(self.config)
 
+    def datajob(self):
```
How do you feel about calling this method either `monitors` or `jobs`?

pashap commented:
Hi @amirwollman, thanks for the PR! I left a few comments, but overall this is awesome. Let me know what you think and we can get this PR merged.

amirwollman committed:
Rewrote Monitors module into monitoring.py,
Changed it's name to Monitors and changed the call for it in __init__.py to monitorsjob instead of datajob.

amirwollman committed:
Corrected Caps.

amirwollman commented:
So I've made the changes, sorry for the mess, I was given a deadline at work to get this working so I didnt quite do a 'clean job' so to speak.. I tried to correct some of my previous errors but if you notice any others let me know and I'll get to work on these as well.

amirwollman commented:
I only now saw your other comment. I've changed the method's name to monitorsjob, (tried to stick to the datafeed pattern)  but this is also changeable if needed.

pashap commented on nsone/__init__.py:
```diff
@@ -85,6 +86,15 @@ def datafeed(self):
+        """
+        Return a new raw REST interface to job resources
+
+        :rtype: :py:class:`nsone.rest.data.Job`
```
Remnant from the move to `monitoring` -> `nsone.rest.monitoring.Monitor`

pashap commented on nsone/__init__.py:
```diff
@@ -85,6 +86,15 @@ def datafeed(self):
         import nsone.rest.data
         return nsone.rest.data.Feed(self.config)
 
+    def monitorsjob(self):
```
I think it makes sense to call this method `monitors`, following the returned classes name. Thoughts?

pashap commented:
@amirwollman I left two small comments, but itll be good to merge after.

amirwollman committed:
Changed method name to "monitors" and corrected explanation comment.

pashap commented:
Thanks for the PR @amirwollman, merging now. No need to worry about the CI fail, fixing that now.
